database/sql/instances_test.go

```diff
@@ -164,6 +179,115 @@ func (s *InstancesTestSuite) TestGetInstanceByNameFetchInstanceFailed() {
 	s.Require().Equal("fetching instance: fetching instance by name: not found", err.Error())
 }
 
+func (s *InstancesTestSuite) TestDeleteInstance() {
+	storeInstance := s.Fixtures.Instances[0]
+	err := s.Store.DeleteInstance(context.Background(), s.Fixtures.Pool.ID, storeInstance.Name)
+
+	s.Require().Nil(err)
+
+	_, err = s.Store.GetPoolInstanceByName(context.Background(), s.Fixtures.Pool.ID, s.Fixtures.Instances[0].Name)
+	s.Require().Equal("fetching instance: fetching pool instance by name: not found", err.Error())
+}
+
+func (s *InstancesTestSuite) TestDeleteInstanceFailed() {
```

Comment:
This mostly tests if ```DeleteInstance()``` validates the pool ID. Should be renamed to something like: ```TestDeleteInstanceInvalidPoolID```

Same with a few more functions bellow.

Reply:
If you want to test a condition where the database returns an error, you have two options:

  * Use mocks
  * Make/open the database read-only and try to mutate the data

Reply:
> This mostly tests if `DeleteInstance()` validates the pool ID. Should be renamed to something like: `TestDeleteInstanceInvalidPoolID`
> 
> Same with a few more functions bellow.

Fixed

Reply:
> If you want to test a condition where the database returns an error, you have two options:
> 
> * Use mocks
> * Make/open the database read-only and try to mutate the data

Sounds good. I will add such a test case in a separate PR. Thank-you for the suggestion.